File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
l = [0] * 999999
result = []
M, N = map(int, input().split())

for i in range(999999):
    if l[i] == 0:
        v = i + 2
        result.append(v)
        for j in range(1, int(1000000 / v) + 1):
            mul = v * j
            l[mul - 2] = 1


result = list(filter(lambda x: M <= x <= N, result))

for r in result:
    print(r)

File: 백준/Silver/1929. 소수 구하기/소수 구하기.js
let fs = require('fs');
let [M, N] = fs.readFileSync(0, 'utf8').trim().split(' ').map(Number);

function solution(M, N) {
  const nums = new Array(N).fill(0).map((_, i) => i + 1);

  nums[0] = '';
  for (let i = 2; i * i <= N; i++) {
    if (nums[i - 1]) {
      let j = 2;
      while (i * j <= N) {
        nums[i * j - 1] = '';
        j += 1;
      }
    }
  }

  return nums.filter((v, i) => i >= M - 1 && v).join('\n');
}

console.log(solution(M, N));
